### PR TITLE
add post comment feature T7

### PR DESCRIPTION
**Pull request by be-part**
Ticket 7 nc_games post comment feature :)

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -163,4 +163,61 @@ describe('GET /api/reviews/:review_id/comments', () => {
...
+  test("Get status 201 response", () => {
+  const newComment = {
+        username: "mallionaire",
+        body: "giant Jenga is better",
+      };
+      return request(app)
+        .post("/api/reviews/2/comments")
+        .send(newComment)
+        .expect(201)
+  })
```
Comment (MKD01) [inline __tests__/app.test.js]: This is also being tested in the below test so doesnt need to be a separate test.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -163,4 +163,61 @@ describe('GET /api/reviews/:review_id/comments', () => {
...
+        body: "giant Jenga is better",
+      };
+  return request(app)
+  .post("/api/reviews/NotAnId/comments")
+        .send(newComment)
+        .expect(400)
+        .then((response) => {
+         expect(response.body.msg).toBe('bad request')
+        })
+  });
```
Comment (MKD01) [inline __tests__/app.test.js:212]: Great tests so far, however there are some cases which have been be tested for yet. 

What should happen if a request was made that was miss some of the required fields in the request body? i.e. no username or body properties.

What should happen if the username given in the request body did not exist?

What should happen if extra properties were given in the req body?

**Review by MKD01 — CHANGES_REQUESTED** [review]
Great work so far, but there are a few things to take a look at, once you have made all your changes push them up to this branch and then re-submit a review request via /nchelp pr

File: __tests__/app.test.js [hunk 3]
```diff
@@ -163,4 +163,104 @@ describe('GET /api/reviews/:review_id/comments', () => {
...
+         .expect(400)
+         .then((response) => {
+          expect(response.body.msg).toBe('bad request')
+         })
+   });
+  test("Return error 400 and message of 'bad request' if message body has extra properties", () => {
+    const newComment = {
+         username: "mallionaire",
+         body: "giant Jenga is better",
+         votes: 7
```
Comment (Jodiech) [inline __tests__/app.test.js]: When extra properties are posted, this should be ignored, as per the ticket: **Note: the status code should be 201 which signifies that something was created**

File: __tests__/app.test.js [hunk 4]
```diff
@@ -163,4 +163,104 @@ describe('GET /api/reviews/:review_id/comments', () => {
...
+        .send(newComment)
+        .expect(400)
+        .then((response) => {
+         expect(response.body.msg).toBe('bad request')
+        })
+  });
+  test("Return error 400 and message of 'bad request' if message body is missing properties", () => {
+    const newComment = {
+         username: "mallionaire",
+       };
```
Comment (Jodiech) [inline __tests__/app.test.js:216]: Great testing for missing required fields 👍

**Review by Jodiech — COMMENTED** [review]
Hey 👋 happy for you to merge the changes you have made for this ticket once you have updated the test for unnecessary properties. 😄

**Review by Beth-McCormack — APPROVED** [review]

**Pull request merged**